src/main/java/com/stripe/net/UntypedMapDeserializer.java

```diff
@@ -0,0 +1,71 @@
+package com.stripe.net;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+
+public class UntypedMapDeserializer {
+  /**
+   * Deserialize JSON into untyped map.
+   * {@code JsonArray} is represented as {@code Object[]}.
```

Comment:
I'd prefer that we use lists instead of arrays everywhere in the library. `List`s are (IMHO) more flexible and easier to reason about, and make the code easier to maintain. See also #605.